projects/dataset_recorder/dataset_recorder.cc

```diff
@@ -1,100 +1,23 @@
-// BoB includes
+#include "robot_recorder.h"
+
+// BoB robotics includes
 #include "common/background_exception_catcher.h"
-#include "common/bn055_imu.h"
-#include "common/map_coordinate.h"
-#include "common/stopwatch.h"
-#include "gps/gps_reader.h"
-#include "imgproc/opencv_unwrap_360.h"
-#include "navigation/image_database.h"
 #include "robots/ackermann/rc_car_bot.h"
 
-#ifdef DUMMY_CAMERA
-#include "video/randominput.h"
-#else
-#include "video/panoramic.h"
-#endif
-
 // Third-party includes
-#include "plog/Log.h"
 #include "third_party/CLI11.hpp"
 
-// POSIX includes
-#include <unistd.h>
-
-// Standard C includes
-#include <ctime>
-
-// Standard C++ includes
-#include <array>
-#include <chrono>
-#include <memory>
-#include <thread>
-
 using namespace BoBRobotics;
-using namespace std::literals;
-using namespace units::angle;
-using namespace units::length;
-using namespace units::literals;
 using namespace units::time;
-
-/*
- * Jetsons sadly don't record the time and date while powered off. We can
- * sometimes get the date from the GPS though (also sadly) we don't always
- * get given this information. So let's have a command-line option to force
- * using the system clock. The onus is then on the user to check that the
- * system clock's date is correct.
- *
- * We do however always get *time* information with GPS coordinates, so we
- * can at least rely on that being correct.
- */
-const std::tm &
-getCurrentDateTime(GPS::GPSReader &gps, const CLI::Option &useSystemClock)
-{
-    auto currentTime = gps.getCurrentDateTime();
-
-    // We already have current date from GPS
-    if (currentTime.tm_year) {
-        goto out;
-    }
-
-    // Get the date from the system clock (the time still comes from the GPS)
-    if (useSystemClock) {
-        LOGW << "Using system clock to get current date. This may not be correct!";
-        const auto rawTime = std::time(nullptr);
-        const auto &systemTime = *gmtime(&rawTime);
-        currentTime.tm_mday = systemTime.tm_mday;
-        currentTime.tm_mon = systemTime.tm_mon;
-        currentTime.tm_year = systemTime.tm_year;
-        goto out;
-    }
-
-    // See if we get a valid date message in the next 5s...
-    {
-        Stopwatch sw;
-        sw.start();
-        while (sw.elapsed() < 5s) {
-            currentTime = gps.getCurrentDateTime();
-            if (currentTime.tm_year) {
-                goto out;
-            }
-        }
-    }
-
-    // ...and if we didn't, give up
-    throw std::runtime_error{ "Timed out waiting for date from GPS" };
-
-    // So sue me.
-out:
-    const auto time = mktime(&currentTime);
-    return *localtime(&time);
-}
+using namespace units::literals;
 
 int
 bobMain(int argc, char *argv[])
 {
     CLI::App app{ "Record data with RC car robot" };
     auto *useSystemClock = app.add_flag("--use-system-clock",
                     "Use the system clock to get current date rather than GPS");
+    app.allow_extras();
```

Comment:
why not just parse runtime as an optional argument?

Reply:
Good point. That would make the logic cleaner too